### PR TITLE
[TEVA-1510] Add reset password functionality

### PR DESCRIPTION
**Pull request by cesidio**
## Jira ticket URL
https://dfedigital.atlassian.net/browse/TEVA-1510

## Review app
https://teaching-vacancies-review-pr-2479.london.cloudapps.digital

## Screenshots of UI changes:
Screenshot 2020-12-04 at 16 56 32
Screenshot 2020-12-04 at 16 57 04
Screenshot 2020-12-04 at 16 57 24
Screenshot 2020-12-04 at 16 58 47
Screenshot 2020-12-04 at 16 59 07
Screenshot 2020-12-04 at 17 01 04
Screenshot 2020-12-04 at 17 03 46
Screenshot 2020-12-04 at 16 56 04

### CODE REVIEW

File: app/controllers/jobseekers/passwords_controller.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  before_action :check_reset_password_token, only: %i[edit]
+  after_action :remove_devise_notice_flash!, only: %i[create]
+
+protected
+
+  def reset_password_token
+    Devise.token_generator.digest(resource_class, :reset_password_token, params[:reset_password_token] || params[resource_name][:reset_password_token])
+  end
+
+  def set_jobseeker_from_token
```
Comment (cpjmcquillan) [inline app/controllers/jobseekers/passwords_controller.rb:12]: I'm surprised we don't get more of this for free

File: config/initializers/devise.rb [hunk 2]
```diff
@@ -133,7 +133,7 @@
   config.send_email_changed_notification = true
 
   # Send a notification email when the user's password is changed.
-  config.send_password_change_notification = true
+  config.send_password_change_notification = false
```
Comment (cpjmcquillan) [inline config/initializers/devise.rb:136]: I thought we wanted this to be true?

File: app/controllers/jobseekers/passwords_controller.rb (hunk 1 above)
Comment (cesidio) [reply app/controllers/jobseekers/passwords_controller.rb:12]: the problem is that at that stage it does not retrieve the model by default. Only when you submit the token and the new password.

File: config/initializers/devise.rb (hunk 2 above)
Comment (cesidio) [reply config/initializers/devise.rb:136]: this is not when we request to change the password, this is when a new password has been set. For the moment we do not have a template for that, but we should probably add it back.

File: spec/support/mailer_helpers.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+module MailerHelpers
+  def last_email
+    ActionMailer::Base.deliveries.last
+  end
+
+  def delivered_emails
```
Comment (cpjmcquillan) [inline spec/support/mailer_helpers.rb:6]: Do you want to update `jobseekers_can_sign_up_to_an_account_spec.rb` please? 🙏

**Review by cpjmcquillan — APPROVED** [review]

**Pull request merged**